embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crash/CrashDataSourceImpl.kt

```diff
@@ -0,0 +1,131 @@
+package io.embrace.android.embracesdk.capture.crash
+
+import io.embrace.android.embracesdk.Severity
+import io.embrace.android.embracesdk.anr.AnrService
+import io.embrace.android.embracesdk.arch.datasource.LogDataSourceImpl
+import io.embrace.android.embracesdk.arch.destination.LogEventData
+import io.embrace.android.embracesdk.arch.destination.LogWriter
+import io.embrace.android.embracesdk.arch.limits.OnOffLimitStrategy
+import io.embrace.android.embracesdk.arch.schema.EmbType
+import io.embrace.android.embracesdk.arch.schema.EmbType.System.ReactNativeCrash.embAndroidReactNativeCrashJsExceptions
+import io.embrace.android.embracesdk.arch.schema.SchemaType
+import io.embrace.android.embracesdk.arch.schema.TelemetryAttributes
+import io.embrace.android.embracesdk.config.ConfigService
+import io.embrace.android.embracesdk.gating.GatingService
+import io.embrace.android.embracesdk.internal.crash.CrashFileMarker
+import io.embrace.android.embracesdk.internal.utils.Uuid.getEmbUuid
+import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
+import io.embrace.android.embracesdk.ndk.NdkService
+import io.embrace.android.embracesdk.opentelemetry.embAndroidThreads
+import io.embrace.android.embracesdk.opentelemetry.embCrashNumber
+import io.embrace.android.embracesdk.opentelemetry.exceptionMessage
+import io.embrace.android.embracesdk.opentelemetry.exceptionStacktrace
+import io.embrace.android.embracesdk.opentelemetry.exceptionType
+import io.embrace.android.embracesdk.opentelemetry.logRecordUid
+import io.embrace.android.embracesdk.payload.JsException
+import io.embrace.android.embracesdk.payload.LegacyExceptionInfo
+import io.embrace.android.embracesdk.prefs.PreferencesService
+import io.embrace.android.embracesdk.session.orchestrator.SessionOrchestrator
+import io.embrace.android.embracesdk.session.properties.EmbraceSessionProperties
+
+/**
+ * Intercept and track uncaught Android Runtime exceptions
+ */
+internal class CrashDataSourceImpl(
+    configService: ConfigService,
+    private val sessionOrchestrator: SessionOrchestrator,
+    private val sessionProperties: EmbraceSessionProperties,
+    private val anrService: AnrService?,
+    private val ndkService: NdkService,
+    @Suppress("UnusedPrivateMember") private val gatingService: GatingService,
+    private val preferencesService: PreferencesService,
+    private val crashMarker: CrashFileMarker,
+    private val logger: InternalEmbraceLogger,
+    private val logWriter: LogWriter,
+) : CrashDataSource,
+    LogDataSourceImpl(
+        destination = logWriter,
+        logger = logger,
+        limitStrategy = OnOffLimitStrategy { configService.autoDataCaptureBehavior.isUncaughtExceptionHandlerEnabled() }
```

Comment:
I'd suggest replacing this with `NoopLimitStrategy` as we don't ever want to limit the capture of uncaught exceptions. Control of data capture can happen in `DataSourceModule`, where this lambda could be supplied as a value to `configGate`

Reply:
@bidetofevil ^

Reply:
Yup, that makes sense! I was wondering why something like this wasn't there - configGate is much better.

Reply:
I replaced with `NoopStrategy` and will move the CrashDataSource into DataSourcesModule in another PR, because dependencies make it hard to do it in an easy way within this PR. At that point, we can put it into `configGate`.